src/Moryx.Products.Management/Implementation/Storage/ProductStorage.cs

```diff
@@ -574,17 +575,21 @@ where links.All(l => l.Id != link.Id)
                             linkEntity = typeEntity.Parts.First(p => p.Id == link.Id);
                         }
                         linkStrategy.SavePartLink(link, linkEntity);
-                        linkEntity.Child = GetPartEntity(uow, link);
+                        linkEntity.Child = GetPartEntity(ref saverContext, link);
                     }
                 }
             }
 
             return typeEntity;
         }
 
-        private ProductTypeEntity GetPartEntity(IUnitOfWork uow, IProductPartLink link)
+        private ProductTypeEntity GetPartEntity(ref ProductPartsSaverContext saverContext, IProductPartLink link)
         {
-            return link.Product.Id == 0 ? SaveProduct(uow, link.Product) : uow.GetEntity<ProductTypeEntity>(link.Product);
+            if (saverContext.EntityCache.ContainsKey((ProductIdentity) link.Product.Identity))
```

Comment:
Using the identity instead of the object for the dictionary is genius! Since it overrides `GetHashcode` this even works if the user did not properly take care of product type objects.